src/interface/solution.jl

```diff
@@ -0,0 +1,87 @@
+@doc raw"""
+    AbstractSolution{T,U<:Integer}
+
+By definitioon,
+"""
+abstract type AbstractSolution{T,U<:Integer} end
+
+@doc raw"""
+    AbstractSample{T,U<:Integer}
+"""
+abstract type AbstractSample{T,U<:Integer} end
+
+@doc raw"""
+    State{U<:Integer}
+"""
+const State{U<:Integer} = AbstractVector{U}
+
+@doc raw"""
+    solution(model)::AbstractSolution{T,U} where {T,U<:Integer}
+
+Returns the model's current solution.
+"""
+function solution end
+
+@doc raw"""
+    sample(model, i::Integer)::AbstractSample{T,U} where {T,U<:Integer}
+
+Returns the ``i``-th sample on the model's current solution, if available.
+"""
+function sample end
+
+@doc raw"""
+    state(sample::AbstractSample{T,U})::AbstractVector{U} where {T,U<:Integer}
+
+    state(model, i::Integer)::AbstractVector{U} where {U<:Integer}
+
+Returns the state vector corresponding to the ``i``-th sample on the model's current solution, if available.
```

Comment:
State vector has a deep meaning in quantum mechanics and QC. How about just sample of bitstring?

Reply:
I see! I don't think we will be changing the name of the function (since it's cool in the field of MH), but the explanation should be rephrased, yes.

Reply:
Done